concrete/src/User/CsvWriter.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+namespace Concrete\Core\User;
+
+use Concrete\Core\Attribute\Category\UserCategory;
+use League\Csv\Writer;
+
+/**
+ * A Writer class for user objects
+ */
+class CsvWriter
+{
+
+    /** @var Writer The writer we use to output */
+    protected $writer;
+
+    /** @var UserCategory */
+    protected $category;
+
+    public function __construct(Writer $writer, UserCategory $category)
+    {
+        $this->writer = $writer;
+        $this->category = $category;
+    }
+
+    /**
+     * Insert a header row for this result set
+     */
+    public function insertHeaders()
+    {
+        $this->writer->insertOne(iterator_to_array($this->getHeaders()));
+    }
+
+    /**
+     * Insert all data from the passed userlist
+     * @param \Concrete\Core\User\UserList $list
+     */
+    public function insertUserList(UserList $list)
+    {
+        $this->writer->insertAll($this->projectList($list));
+    }
+
+    /**
+     * A generator that takes a UserList and converts it to CSV rows
+     * @param \Concrete\Core\User\UserList $list
+     * @return \Generator
+     */
+    private function projectList(UserList $list)
+    {
+        $statement = $list->deliverQueryObject()->execute();
+
+        foreach ($statement as $result) {
+            $user = $list->getResult($result);
+            yield iterator_to_array($this->projectUser($user));
+        }
+    }
+
+    /**
+     * Turn a user into an array
+     * @param \Concrete\Core\User\UserInfo $user
+     * @return array
+     */
+    private function projectUser(UserInfo $user)
+    {
+        yield $user->getUserName();
+        yield $user->getUserEmail();
+
+        $userRegistrationDate = $user->getUserDateAdded();
+        yield $userRegistrationDate->format('d/m/Y G:i');
```

Comment:
Thinking a bit more about this: we may have custom date/time attributes, and I'd render dates in a consistent way: [here's](https://github.com/concrete5/concrete5/blob/75e7eca71a69752a6f32853db0699579cb858410/concrete/src/Entity/Attribute/Value/Value/DateTimeValue.php#L42) how date/time attribute values are rendered